File: 11-ContainerWithMostWater/11-ContainerWithMostWater.js
// Last updated: 24/07/2026, 23:55:33
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    let maxArea = 0; i = 0; j = height.length-1;

    while (i < j) {
        if (height[i] < height[j]){
            maxArea = Math.max(maxArea, height[i] * (j - i));
            i++
        }else{
            maxArea = Math.max(maxArea, height[j] * (j - i));
            j--
        }
    }


    return maxArea
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 24/07/2026, 23:55:41
class Solution {
    public int maxArea(int[] height) {
        int max=0;
        int temp=0;
        int i=0;
        int j=height.length-1;
        while(i<j){
            temp = Math.min(height[i],height[j]) * (j-i);
            if(height[i]<height[j]){
                i++;
            }else{
                j--;
            }

            if(temp>max){
                max=temp;
            }
        }

        return max;
    }
}